core/events.js

```diff
@@ -347,879 +353,6 @@ Blockly.Events.fromJson = function(json, workspace) {
-};
-
-/**
- * Returns the parentId and input if the block is connected,
- *   or the XY location if disconnected.
- * @return {!Object} Collection of location info.
- * @private
- */
-Blockly.Events.Move.prototype.currentLocation_ = function() {
-  var workspace = Blockly.Workspace.getById(this.workspaceId);
-  var block = workspace.getBlockById(this.blockId);
-  var location = {};
-  var parent = block.getParent();
-  if (parent) {
-    location.parentId = parent.id;
-    var input = parent.getInputWithBlock(block);
-    if (input) {
-      location.inputName = input.name;
-    }
-  } else {
-    location.coordinate = block.getRelativeToSurfaceXY();
-  }
-  return location;
-};
-
-/**
- * Does this event record any change of state?
- * @return {boolean} True if something changed.
- */
-Blockly.Events.Move.prototype.isNull = function() {
-  return this.oldParentId == this.newParentId &&
-      this.oldInputName == this.newInputName &&
-      goog.math.Coordinate.equals(this.oldCoordinate, this.newCoordinate);
-};
-
-/**
- * Run a move event.
- * @param {boolean} forward True if run forward, false if run backward (undo).
- */
-Blockly.Events.Move.prototype.run = function(forward) {
-  var workspace = this.getEventWorkspace_();
-  var block = workspace.getBlockById(this.blockId);
-  if (!block) {
-    console.warn("Can't move non-existant block: " + this.blockId);
-    return;
-  }
-  var parentId = forward ? this.newParentId : this.oldParentId;
-  var inputName = forward ? this.newInputName : this.oldInputName;
-  var coordinate = forward ? this.newCoordinate : this.oldCoordinate;
-  var parentBlock = null;
-  if (parentId) {
-    parentBlock = workspace.getBlockById(parentId);
-    if (!parentBlock) {
-      console.warn("Can't connect to non-existant block: " + parentId);
-      return;
-    }
-  }
-  if (block.getParent()) {
-    block.unplug();
-  }
-  if (coordinate) {
-    var xy = block.getRelativeToSurfaceXY();
-    block.moveBy(coordinate.x - xy.x, coordinate.y - xy.y);
-  } else {
-    var blockConnection = block.outputConnection || block.previousConnection;
-    var parentConnection;
-    if (inputName) {
-      var input = parentBlock.getInput(inputName);
-      if (input) {
-        parentConnection = input.connection;
-      }
-    } else if (blockConnection.type == Blockly.PREVIOUS_STATEMENT) {
-      parentConnection = parentBlock.nextConnection;
-    }
-    if (parentConnection) {
-      blockConnection.connect(parentConnection);
-    } else {
-      console.warn("Can't connect to non-existant input: " + inputName);
-    }
-  }
-};
-
-/**
- * Class for a block drag event. Fired when block dragged into or out of
- * the blocks UI.
- * @param {Blockly.Block} block The moved block.  Null for a blank event.
- * @extends {Blockly.Events.Abstract}
- * @constructor
- */
-Blockly.Events.DragOutside = function(block) {
-  if (!block) {
-    return;  // Blank event to be populated by fromJson.
-  }
```

Comment:
I noticed that this check/early exit got removed in the refactor, is it no longer necessary?

Reply:
Added back in, for this and EndBlockDrag.